### DISCUSSION TITLE
Feat: Mask Input Machine

### DISCUSSION THREAD

**Discussion by juliomuhlbauer** [chakra-ui/zag#1629] [Ideas]
A mask input machine alongside number input.
Eg: phone, documents, zip code
Given a mask: (eg: mask="99-999999"), it would format accordingly.

There are not some agnostic good libs out here ( I have already tried some).

I wasusing the number input machine provided by zag with some workarounds. The number input machine had the ability to provide a function to format and parse the number. It worked with some regex. But now I have to use the methods provided by the Inlt.NumberFormat. I wanted to give a custom mask.

**Comment by juliomuhlbauer** [comment]
Example: https://primereact.org/inputmask/

**Comment by segunadebayo** [comment]
We dont intend to implement this.
